extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -347,7 +347,7 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 
 	public async searchJupyterBooks(treeItem?: BookTreeItem): Promise<void> {
 		let folderToSearch: string;
-		if (treeItem && treeItem.book.type !== BookTreeItemType.Notebook) {
+		if (treeItem && treeItem.sections?.length) {
```

Comment:
Is the intent to check that length is defined? And if so would you consider a length of 0 to be valid? Because right now it's going to skip this section if it's undefined OR it's 0. So a `=== undefined` check would probably be more appropriate here.

Reply:
the intent is to check if there are sections (and therefore there are children in the tree)

Reply:
I guess my question is - is a sections length of 0 ever valid?